src/Raven.Server/Documents/ETL/Providers/RelationalDatabase/Snowflake/RelationalWriters/SnowflakeDatabaseWriter.cs

```diff
@@ -0,0 +1,138 @@
+using Raven.Server.NotificationCenter.Notifications;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Snowflake.Data.Client;
+using Sparrow.Json;
+using DbCommandBuilder = Raven.Server.Documents.ETL.Providers.RelationalDatabase.Common.RelationalWriters.DbCommandBuilder;
+
+namespace Raven.Server.Documents.ETL.Providers.RelationalDatabase.Snowflake.RelationalWriters;
+
+public class SnowflakeDatabaseWriter: RelationalDatabaseWriterBase<SnowflakeConnectionString, SnowflakeEtlConfiguration>
+{
+    private const string SnowflakeEtlTag = "Snowflake ETL";
+
+    private readonly SnowflakeEtlConfiguration _snowflakeEtlConfiguration;
+
+    public SnowflakeDatabaseWriter(DocumentDatabase database, SnowflakeEtlConfiguration configuration, RelationalDatabaseEtlMetricsCountersManager sqlMetrics,
+        EtlProcessStatistics statistics, bool shouldConnectToTarget = true) : base(database, configuration, sqlMetrics, statistics, shouldConnectToTarget)
+    {
+        _snowflakeEtlConfiguration = configuration;
+    }
+
+    public override bool ParametrizeDeletes => false;
+
+    protected override string GetConnectionString(EtlConfiguration<SnowflakeConnectionString> configuration)
+    {
+        return configuration.Connection.ConnectionString;
+    }
+
+    protected override DbCommandBuilder GetInitializedCommandBuilder()
+    {
+        return new DbCommandBuilder { Start =  string.Empty, End = string.Empty };
+    }
+
+    protected override DbProviderFactory GetDbProviderFactory(EtlConfiguration<SnowflakeConnectionString> configuration)
+    {
+        return SnowflakeDbFactory.Instance;
+    }
+
+    protected override void CreateAlertCannotOpenConnection(DocumentDatabase database, string etlConfigurationName, string connectionStringName, Exception e)
+    {
+        database.NotificationCenter.Add(AlertRaised.Create(
+            database.Name,
+            SnowflakeEtlTag,
+            $"[{etlConfigurationName}] Could not open connection using '{connectionStringName}' connection string",
+            AlertType.SnowflakeEtl_ConnectionError,
+            NotificationSeverity.Error,
+            key: $"{etlConfigurationName}/{connectionStringName}",
+            details: new ExceptionDetails(e)));
+    }
+
+    protected override int? GetCommandTimeout()
+    {
+        return _snowflakeEtlConfiguration.CommandTimeout;
+    }
+
+    protected override bool ShouldQuoteTables()
+    {
+        return false;
+    }
+
+    protected override string GetParameterNameForDbParameter(string paramName)
+    {
+        return paramName;
+    }
+
+    protected override string GetParameterNameForCommandString(string targetParamName, bool parseJson)
+    {
+        return parseJson ? $"PARSE_JSON(:{targetParamName})" : $":{targetParamName}";
+    }
+
+    protected override void EnsureParamTypeSupportedByDbProvider(DbParameter parameter)
+    {
+        if (parameter.DbType == DbType.AnsiString)
```

Comment:
By default, `DbParameter` has `DbType.AnsiString`. `SetParamValue` changes it, so we don't care
At `null` value, we can't update `DbType`, because there's no `DbType.Null`. DbType stays `AnsiString`. Snowflake doesn't support AnsiString - we need to correct that to `DbType.String`. Only Snowflake has this, so we need granular code, to not break Sql


Will add also a comment in code similar to this ^

Reply:
Of course, there may be other hiccups which I don't know about yet, which will set `DbType.AnsiString` and break ETL. This way we're guarding that case.

Reply:
ok